pipelines/src/main/scala/dagr/pipelines/ContaminateAndEvaluateVBID.scala

```diff
@@ -0,0 +1,135 @@
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package dagr.pipelines
+
+import java.nio.file.{Files, Path}
+
+import dagr.core.cmdline.Pipelines
+import dagr.core.execsystem.{Cores, Memory}
+import dagr.core.tasksystem.{EitherTask, Pipeline, SimpleInJvmTask}
+import dagr.sopt.{arg, clp}
+import dagr.tasks.DagrDef._
+import dagr.tasks.misc.LinkFile
+import dagr.tasks.picard.{CollectHsMetrics, DownsampleSam, DownsamplingStrategy, MergeSamFiles}
+import htsjdk.samtools.SAMFileHeader.SortOrder
+
+import scala.collection.mutable
+
+/**
+  * Downsample Bams to create composite contaminated bams,
+  * then downsample to various depths
+  * and run VBID with different number of target snps to evaluate
+  * dependance of VBID on contamination & depth & number of target snps
+  */
+@clp(description = "Example FASTQ to BAM pipeline.", group = classOf[Pipelines])
+class ContaminateAndEvaluateVBID
+(@arg(flag = "i", doc = "Input bams.") val bams: Seq[PathToBam],
+ @arg(flag = "r", doc = "Reference fasta.") val ref: PathToFasta,
+ @arg(flag = "o", doc = "Output directory.") val out: DirPath,
+ @arg(flag = "t", doc = "Target regions.") val targets: PathToIntervals,
+ @arg(flag = "p", doc = "Output file prefix.") val prefix: String,
+ @arg(flag = "c", doc = "Contamination levels to evaluate.") val contaminations: Seq[Double],
+ @arg(flag = "d", doc = "Depth values to evaluate.") val depths: Seq[Integer],
+ @arg(flag = "m", doc = "Number of markers to use.") val markers: Seq[Integer]
+
+) extends Pipeline(Some(out)) {
+
+  override def build(): Unit = {
+    val strat = Some(DownsamplingStrategy.HighAccuracy)
+
+    val bam = out.resolve(prefix + ".bam")
+    val tmpBam = out.resolve(prefix + ".tmp.bam")
+    val metricsPrefix: Some[DirPath] = Some(out.resolve(prefix))
+    Files.createDirectories(out)
+    val bamYield = new mutable.HashMap[PathToBam, Int]
+
+    bams.map(bam => {
+      val metricPath: Path = out.resolve(prefix + bam.getFileName + ".qualityYieldMetrics")
```

Comment:
Can the suffix be the same one that CollectMultipleMetrics would produce?  Then MultiQC can be used too!

Reply:
Why is this `qualityYieldMetrics` when the output is presumably from HsMetrics?